pkg/controller/migstorage/validation.go

```diff
@@ -168,73 +161,79 @@ func (r ReconcileMigStorage) validateVolumeSnapshotStorage(storage *migapi.MigSt
 	// Secret
 	secret, err := storage.GetVolumeSnapshotCredSecret(r)
 	if err != nil {
-		return liberr.Wrap(err)
+		return liberr.Wrap(fmt.Errorf("error getting migstorage backup provider secret %s/%s: %v",
+			storage.Spec.VolumeSnapshotConfig.CredsSecretRef.Namespace,
+			storage.Spec.VolumeSnapshotConfig.CredsSecretRef.Name,
+			err))
 	}
 
 	if storage.Spec.VolumeSnapshotProvider != "" {
 		// Unknown provider.
 		if provider == nil {
-			storage.Status.SetCondition(migapi.Condition{
+			storage.Status.SetAndLogCondition(log, migapi.Condition{
 				Type:     InvalidVSProvider,
 				Status:   True,
 				Reason:   NotSupported,
 				Category: Critical,
-				Message:  InvalidVSProviderMessage,
-			})
+				Message:  fmt.Sprintf("could not get provider for spec.VolumeSnapshotProvider: %s", storage.Spec.VolumeSnapshotProvider),
+			}, storage.Namespace, storage.Name)
 			return nil
 		}
 
 		// NotSet
 		if !migref.RefSet(settings.CredsSecretRef) {
-			storage.Status.SetCondition(migapi.Condition{
+			storage.Status.SetAndLogCondition(log, migapi.Condition{
 				Type:     InvalidVSCredsSecretRef,
 				Status:   True,
 				Reason:   NotSet,
 				Category: Critical,
-				Message:  InvalidVSCredsSecretRefMessage,
-			})
+				Message:  "spec.VolumeSnapshotConfig.CredsSecretRef name or namespace is empty",
+			}, storage.Namespace, storage.Name)
 			return nil
 		}
 
 		// NotFound
 		if secret == nil {
-			storage.Status.SetCondition(migapi.Condition{
+			storage.Status.SetAndLogCondition(log, migapi.Condition{
 				Type:     InvalidVSCredsSecretRef,
 				Status:   True,
 				Reason:   NotFound,
 				Category: Critical,
-				Message:  InvalidVSCredsSecretRefMessage,
-			})
+				Message: fmt.Sprintf("VolumeSnapshotConfig secret %s/%s not found",
```

Comment:
These messages provide good context, I like that a lot!

Reply:
I kind of disagree.  The condition message seem redundant with the values in the Spec.

Reply:
Also, can we please use good grammar in the messages.  Begin with a capital and end with a period?

Reply:
> I kind of disagree. The condition message seem redundant with the values in the Spec.

The goal of this is to bubble out the state that was processed in the controller. This is useful in scenarios where:

1. For some reason the mig-controller pod is not in running state.
2. User changed the spec, but the condition is still set due to the old spec value.

In the above case, as a user I expect the controller to explicitly tell me which secret it was unable to find